actions/auth-pubtkt

```diff
@@ -0,0 +1,138 @@
+    gen_tkt.add_argument('--tokens',
+                         help='tokens, usually containing the user groups')
+
+    subparsers.required = True
+    return parser.parse_args()
+
+
+def subcommand_create_key_pair(_):
+    """Create public/private key pair for signing the auth_pubtkt
+    tickets.
+    """
+    private_key_file = os.path.join(KEYS_DIRECTORY, 'privkey.pem')
+    public_key_file = os.path.join(KEYS_DIRECTORY, 'pubkey.pem')
+
+    os.path.exists(KEYS_DIRECTORY) or os.mkdir(KEYS_DIRECTORY)
+
+    if not all([
+            os.path.exists(key_file)
+            for key_file in [public_key_file, private_key_file]
+    ]):
+        pkey = crypto.PKey()
+        pkey.generate_key(crypto.TYPE_DSA, 1024)
+
+        with open(private_key_file, 'w') as priv_key_file:
+            priv_key = crypto.dump_privatekey(crypto.FILETYPE_PEM,
+                                              pkey).decode()
+            priv_key_file.write(priv_key)
+
+        with open(public_key_file, 'w') as pub_key_file:
+            pub_key = crypto.dump_publickey(crypto.FILETYPE_PEM, pkey).decode()
+            pub_key_file.write(pub_key)
+
+        for fil in [public_key_file, private_key_file]:
+            os.chmod(fil, 0o440)
+
+
+def create_ticket(pkey, uid, validuntil, ip=None, tokens=None,
+                  udata=None, graceperiod=None, extra_fields=None):
+    """Create and return a signed mod_auth_pubtkt ticket."""
+    fields = [
+        'uid={}'.format(uid),
+        'validuntil={}'.format(validuntil, type='d'),
+        ip and 'cip={}'.format(ip),
+        tokens and 'tokens={}'.format(tokens),
+        graceperiod and 'graceperiod={}'.format(graceperiod, type='d'),
+        udata and 'udata={}'.format(udata),
+        extra_fields and ';'.join(
+            ['{k}={v}'.format(k, v) for k, v in extra_fields])
+    ]
+    data = ';'.join(filter(None, fields))
+    signature = 'sig={}'.format(sign(pkey, data))
+    return ';'.join([data, signature])
+
+
+def sign(pkey, data):
+    """Calculates and returns ticket's signature."""
+    from OpenSSL import crypto
+    sig = crypto.sign(pkey, data, 'sha1')
+    return base64.b64encode(sig).decode()
+
+
+def subcommand_generate_ticket(arguments):
+    """Generate a mod_auth_pubtkt ticket using login credentials."""
+    from OpenSSL import crypto
+    uid = arguments.uid
+    private_key_file = arguments.private_key_file
+    tokens = arguments.tokens
+    with open(private_key_file, 'r') as fil:
+        pkey = crypto.load_privatekey(crypto.FILETYPE_PEM, fil.read().encode())
+    valid_until = minutes_from_now(30)
+    grace_period = minutes_from_now(25)
+    print(create_ticket(
+        pkey, uid, valid_until, tokens=tokens, graceperiod=grace_period))
+
+
+def minutes_from_now(minutes):
+    """Return a timestamp at the given number of minutes from now."""
+    return (datetime.datetime.now() + datetime.timedelta(minutes=30)).timestamp()
```

Comment:
The default value for minutes should go in the def line. (Otherwise it is just being assigned to 30 here, regardless of the value given.)